src/components/GestureComponents.tsx

```diff
@@ -1,63 +1,153 @@
-/* eslint-disable eslint-comments/no-unlimited-disable */
-/* eslint-disable */
-// @ts-nocheck TODO(TS) provide types
 import React from 'react';
-import ReactNative from 'react-native';
+import {
+  ScrollView as RNScrollView,
+  ScrollViewProps as RNScrollViewProps,
+  Switch as RNSwitch,
+  SwitchProps as RNSwitchProps,
+  TextInput as RNTextInput,
+  TextInputProps as RNTextInputProps,
+  DrawerLayoutAndroid as RNDrawerLayoutAndroid,
+  DrawerLayoutAndroidProps as RNDrawerLayoutAndroidProps,
+  FlatList as RNFlatList,
+  FlatListProps as RNFlatListProps,
+  FlatListProps,
+} from 'react-native';
 
 import createNativeWrapper from '../handlers/createNativeWrapper';
 
-const MEMOIZED = new WeakMap();
+import { NativeViewGestureHandlerProperties } from '../handlers/NativeViewGestureHandler';
 
-function memoizeWrap(Component, config) {
-  if (Component == null) {
-    return null;
-  }
+const MEMOIZED = new WeakMap<
+  React.ComponentType<any>,
+  React.ForwardRefExoticComponent<any>
+>();
+
+function memoizeWrap<P>(
+  Component: React.ComponentType<P>,
+  config?: Readonly<NativeViewGestureHandlerProperties>
+): React.ComponentType<
+  P & NativeViewGestureHandlerProperties & React.RefAttributes<any>
+> {
   let memoized = MEMOIZED.get(Component);
   if (!memoized) {
-    memoized = createNativeWrapper(Component, config);
+    memoized = createNativeWrapper<P>(Component, config);
     MEMOIZED.set(Component, memoized);
   }
   return memoized;
 }
 
-module.exports = {
+const GestureComponentWrappers = {
   /* RN's components */
   get ScrollView() {
-    return memoizeWrap(ReactNative.ScrollView, {
+    return memoizeWrap<RNScrollViewProps>(RNScrollView, {
       disallowInterruption: true,
       shouldCancelWhenOutside: false,
     });
   },
   get Switch() {
-    return memoizeWrap(ReactNative.Switch, {
+    return memoizeWrap<RNSwitchProps>(RNSwitch, {
       shouldCancelWhenOutside: false,
       shouldActivateOnStart: true,
       disallowInterruption: true,
     });
   },
   get TextInput() {
-    return memoizeWrap(ReactNative.TextInput);
+    return memoizeWrap<RNTextInputProps>(RNTextInput);
   },
   get DrawerLayoutAndroid() {
-    const DrawerLayoutAndroid = memoizeWrap(ReactNative.DrawerLayoutAndroid, {
-      disallowInterruption: true,
-    });
-    DrawerLayoutAndroid.positions = ReactNative.DrawerLayoutAndroid.positions;
+    const DrawerLayoutAndroid = memoizeWrap<RNDrawerLayoutAndroidProps>(
+      RNDrawerLayoutAndroid,
+      {
+        disallowInterruption: true,
+      }
+    );
+    // we use literal object since TS gives error when using RN's `positions`
+    // @ts-ignore FIXME(TS) maybe this should be removed?
+    DrawerLayoutAndroid.positions = { Left: 'left', Right: 'right' };
     return DrawerLayoutAndroid;
   },
-  get FlatList() {
-    if (!MEMOIZED.FlatList) {
+
+  // @ts-ignore xddddd
```

Comment:
🤔

Reply:
This was added in WIP commit, will be fixed 😄